main.go

```diff
@@ -30,7 +31,20 @@ func main() {
 	}
 
 	logLevel, _ := flag.CommandLine.GetString("loglevel")
-	initLog(logLevel, os.Stderr)
+	logdir, _ := flag.CommandLine.GetString("logdir")
+	logFileName, _ := flag.CommandLine.GetString("logfile")
+
+	if logFileName == "-" {
+		initLog(logLevel, os.Stderr)
+	} else {
+		logFilePath := path.Join(logdir, logFileName)
+		logFile, logFileErr := openLogFile(logFilePath)
+		if logFileErr != nil {
+			initLog(logLevel, os.Stderr)
+			log.WithError(logFileErr).Fatalf("Failed to open log file %s", logFilePath)
```

Comment:
I know that `Fatal` will internally log and do a `os.Exit(1)`. I think `Fatalf` does the same. If so, the program will exit. Please check.